src/analysis/normed_space/spectrum.lean

```diff
@@ -87,11 +88,29 @@ metric.is_compact_of_is_closed_bounded (is_closed a) (is_bounded a)
 
 theorem spectral_radius_le_nnnorm (a : A) :
   spectral_radius 𝕜 a ≤ ∥a∥₊ :=
+by { refine bsupr_le (λ k hk, _), exact_mod_cast norm_le_norm_of_mem hk }
+
+open ennreal polynomial
+
+theorem spectral_radius_le_pow_nnnorm_pow_one_div (a : A) (n : ℕ) :
+  spectral_radius 𝕜 a ≤ ∥a ^ (n + 1)∥₊ ^ (1 / (n + 1) : ℝ) :=
 begin
-  suffices h : ∀ k ∈ σ a, (∥k∥₊ : ℝ≥0∞) ≤ ∥a∥₊,
-  { exact bsupr_le h, },
-  { intros _ hk,
-    exact_mod_cast norm_le_norm_of_mem hk },
+  refine bsupr_le (λ k hk, _),
+  /- apply easy direction of the spectral mapping theorem for polynomials -/
+  have pow_mem : k ^ (n + 1) ∈ σ (a ^ (n + 1)),
+    by simpa only [one_mul, algebra.algebra_map_eq_smul_one, one_smul, aeval_monomial, one_mul,
+      eval_monomial] using subset_polynomial_aeval a (monomial (n + 1) (1 : 𝕜)) ⟨k, hk, rfl⟩,
+  /- power of the norm is bounded by norm of the power -/
+  have nnnorm_pow_le : ↑(∥k∥₊ ^ (n + 1)) ≤ ↑∥a ^ (n + 1)∥₊,
```

Comment:
```suggestion
  have nnnorm_pow_le : (↑(∥k∥₊ ^ (n + 1)) : ℝ≥0∞) ≤ ↑∥a ^ (n + 1)∥₊,
```
You'll have seen an odd second goal appearing and disappearing in your proof, I believe this comes from Lean not knowing what you wanted coerce to in this line.

Reply:
yes, I did notice that, but then they magically went away so I didn't worry about it. Good to know.